chia/full_node/mempool_check_conditions.py

```diff
@@ -58,7 +58,7 @@ def get_name_puzzle_conditions(
             return NPCResult(uint16(err), None, uint64(0))
         else:
             return NPCResult(None, result, uint64(result.cost + size_cost))
-    except BaseException as e:
+    except Exception as e:
```

Comment:
right, my main concern with this change was that perhaps `AssertionError` would no longer be caught, but as far as I can tell, it would. (the documentation on the built-in exception types are thin on what they inherit from). I can't imagine this was deliberately made to catch `SysExit` or `KeyboardInterrupt`

Reply:
@richardkiss what do you think about this change?

Reply:
> right, my main concern with this change was that perhaps `AssertionError` would no longer be caught, but as far as I can tell, it would. (the documentation on the built-in exception types are thin on what they inherit from).

https://docs.python.org/3.10/library/exceptions.html#exception-hierarchy and also:

```pycon
>>> AssertionError.__mro__
(<class 'AssertionError'>, <class 'Exception'>, <class 'BaseException'>, <class 'object'>)
```

> I can't imagine this was deliberately made to catch `SysExit` or `KeyboardInterrupt`

That's what makes broad exception catching bad.  But I think at some point pyo3 did raise something for us that did inherit from `BaseException` and not `Exception` and so _all_ exceptions got caught up.

Reply:
If we have concerns that we really do need to catch `BaseException`, we could catch `Exception` first and process it as we do now (by logging), then `BaseException` and log it specially, then re-raise. That way we would have a log trail to try to eliminate `BaseException`. We should also probably comment here that it's a temporary measure to be removed once we're confident there are no weird exceptions coming from native libraries.

Reply:
I think I'll pull this out to a separate PR